### ISSUE TITLE
Build JVM and Android libs tasks do not run as part of the standard build process

### ISSUE THREAD

**Issue by kirillzh** [bitcoindevkit/bdk-kotlin#68]
### Description

Currently, when running tests directly from CLI or an IDE, we don't execute `buildJvmLibs` and `buildAndroidLibs` tasks automatically. Usually codegen-like tasks similar to `buildJvmLibs` and `buildAndroidLibs` are executed automatically before `assemble` task is executed, it would be nice to set that up for this project as well.

Pros:
1. No need to manually run `buildJvmLibs` or `buildAndroidLibs` when building project (which happens as part of running tests for example)
2. Easier to develop when making changes to bindings as they changes will be picked up automatically which avoids manually running `buildJvmLibs` and `buildAndroidLibs`.

Cons:
1. I didn't get a chance to test that out but it would be good that `buildJvmLibs` and `buildAndroidLibs` tasks can be cachable to avoid unnecessary execution, since we would be trying to run the task on each build.

### How to Reproduce

Steps:
1. Delete generated bindings in `jvm/src/main/kotlin/org/bitcoindevkit`
2. Run `./gradlew :jvm:test`
3. Fails since native bindings were not auto generated, requires manually running `buildJvmLibs`
4. Same for the `:android` project

### Expected Result

The custom `buildJvmLibs` and `buildAndroidLibs` tasks are executed automatically when assembling `:jvm` and `:android` projects.

### Version

Latest `0.8.0-SNAPSHOT`.

### Artifact

- [X] bdk-jvm
- [X] bdk-android

### Platform

- [ ] ARM64 Android
- [ ] 64-bit x86 Android
- [ ] 32-bit x86 Android
- [ ] 64-bit x86 Linux (kernel 2.6.32+, glibc 2.11+)
- [ ] 64-bit x86 macOS (10.7+, Lion+)
- [ ] ARM64 macOS (11.0+, Big Sur+)

**Comment by kirillzh** [comment]
One potential solution is to make `KotlinCompile` task depends on `buildJvmLib`/`buildAndroidLib`:

```kotlin
// jvm/build.gradle.kts
tasks.withType<KotlinCompile> {
  dependsOn("buildJvmLib")
}

// android/build.gradle.kts
tasks.withType<KotlinCompile> {
  dependsOn("buildAndroidLib")
}
```

**Comment by notmandatory** [comment]
Concept ACK, sounds like a good idea. The `buildJvmLib` and `buildAndroidLib` won't rebuild anything if there were no changes to the `bdk-ffi` submodule, so shouldn't slow things down running them multiple times.

**Comment by thunderbiscuit** [comment]
Agreed. @kirillzh do you want to take this on? If indeed your proposed solution works it will be an easy merge.
